dashboard/src/components/api/workspace/che-workspace.factory.ts

```diff
@@ -401,21 +399,51 @@ export class CheWorkspace {
   /**
    * Starts the given workspace by specifying the ID and the environment name
    * @param workspaceId the workspace ID
-   * @param envName the name of the environment (optional)
-   * @returns {ng.IPromise<any>} promise
    */
-  startWorkspace(workspaceId: string, envName?: string): ng.IPromise<any> {
+  startWorkspace(workspaceId: string): ng.IPromise<any>;
+  /**
+   * Starts the given workspace by specifying the ID and the environment name
+   * @param workspaceId the workspace ID
+   * @param isDebugMode
+   */
+  startWorkspace(workspaceId: string, isDebugMode: boolean): ng.IPromise<any>;
+  /**
+   * Starts the given workspace by specifying the ID and the environment name
+   * @param workspaceId the workspace ID
+   * @param envName the name of the environment
+   */
+  startWorkspace(workspaceId: string, envName: string): ng.IPromise<any>;
+  /**
+   * Starts the given workspace by specifying the ID and the environment name
+   * @param workspaceId the workspace ID
+   * @param envName the name of the environment
+   * @param isDebugMode
+   */
+  startWorkspace(workspaceId: string, envName: string, isDebugMode: boolean): ng.IPromise<any>;
+  startWorkspace(workspaceId: string, envNameOrIsDebugMode?: string | boolean, isDebugMode?: boolean): ng.IPromise<any> {
```

Comment:
This method overloading seems too bulky to me. My proposal is to declare `startWorkspace` in following way

```typescript
startWorkspace(workspaceId: string, isDebugMode: boolean = false, envName?: string) {...}
```

Reply:
done